fog/view/server/tests/router_smoke_tests.rs

```diff
@@ -0,0 +1,139 @@
+        .new_ingest_invocation(None, &ingress_key, &egress_public_key, 0)
+        .unwrap();
+
+    let mut expected_records = Vec::new();
+    const BLOCK_COUNT: u64 = 5;
+    for i in 0..BLOCK_COUNT {
+        let (block, records) = random_block(&mut rng, i, 2);
+        db.add_block_data(&invoc_id1, &block, 0, &records).unwrap();
+        expected_records.extend(records);
+    }
+
+    // Wait until first server has added stuff to ORAM. Since all view servers
+    // should load ORAM at the same time, we could choose to wait for any view
+    // server.
+    let mut allowed_tries = 1000usize;
+    loop {
+        let db_num_blocks = db
+            .get_highest_known_block_index()
+            .unwrap()
+            .map(|v| v + 1) // convert index to count
+            .unwrap_or(0);
+        let server_num_blocks = test_environment
+            .store_servers
+            .as_ref()
+            .unwrap()
+            .iter()
+            .map(|server| server.highest_processed_block_count())
+            .max()
+            .unwrap_or_default();
+        if server_num_blocks > db_num_blocks {
+            panic!(
+                "Server num blocks should never be larger than db num blocks: {} > {}",
+                server_num_blocks, db_num_blocks
+            );
+        }
+        if server_num_blocks == db_num_blocks {
+            log::info!(logger, "Stopping, block {}", server_num_blocks);
+            break;
+        }
+        log::info!(
+            logger,
+            "Waiting for server to catch up to db... {} < {}",
+            server_num_blocks,
+            db_num_blocks
+        );
+        if allowed_tries == 0 {
+            panic!("Server did not catch up to database!");
+        }
+        allowed_tries -= 1;
+        sleep(Duration::from_secs(1));
+    }
+
+    let router_client = test_environment.router_client.as_mut().unwrap();
+    let result =
+        mc_fog_view_server_test_utils::assert_e_tx_out_records(router_client, &expected_records)
+            .await;
+
+    assert!(result.is_ok());
+
+    let result = result.unwrap();
+    // TODO: see what we should do about collating these fields... Right now they
+    // are in sync, but that won't always be the case...
+    assert_eq!(result.highest_processed_block_count, 5);
+    assert_eq!(result.next_start_from_user_event_id, 1);
+    assert_eq!(result.rng_records.len(), 1);
+    assert_eq!(result.rng_records[0].pubkey, egress_public_key);
+    assert_eq!(result.missed_block_ranges.len(), 0);
+    assert_eq!(result.last_known_block_count, 5);
+}
+
+#[test]
+fn test_512() {
+    let (logger, _global_logger_guard) = create_app_logger(o!());
+    const OMAP_CAPACITY: u64 = 512;
+    const STORE_COUNT: usize = 5;
+    let mut test_environment =
+        RouterTestEnvironment::new(OMAP_CAPACITY, STORE_COUNT, logger.clone());
+
+    block_on(test_router_integration(
+        &mut test_environment,
+        logger.clone(),
+    ))
+}
+
+#[test]
+fn test_1_million() {
```

Comment:
What results do you get from `test_1_million` that you don't get from `test_512`? Is `1024*1024` enough OMAP capacity that it wouldn't fit in a single Store? If not, should we talk about adding a test which does exceed the single-store capacity (probably in a future PR)?

Reply:
I'm not exactly sure to be honest. This is what's being tested in the current fog view `smoke_test` is testing, but agree that it might be worth it to add a test that exceeds the single store capacity.